Small update habitatstreams to v1.7 + fileman_up to stop when target not found

cecileherr opened the pull request:
-  A minor update to read_habitatdata.R > **read_habitatstreams** to set the new version (v1.7 2020) as standard version

I made a preliminary controle within n2khab-preprocessing ([html notebook](https://github.com/inbo/n2khab-preprocessing/files/5997216/habitatstreams.zip) or script soon under src/miscellaneous): since the structure of version 1.7 has not changed at all since version 1.6, we do not need to adapt to the script itself.

Please note I have not added any version control yet on purpose (technically not needed to use v1.7, and I suppose the priority is to make the new layers available to the users - we can tackle the version control at a later stage)

Question to users: would it be useful to adapt the letter case of the river names (some are all uppercase, others only start with a capital letter)?

- Update **fileman_up** to stop the script if the file or folder we are searching for is not found and show a meaningful  error message. 

This should be a solution for issue #106

florisvdh commented on R/filemanagement.R:
```diff
@@ -374,7 +374,7 @@ fileman_up <- function(name,
             normalizePath()
     } else {
-        NULL
+        stop(name, " was not found. Searched up to ", path)
```
```r
> fileman_up("tralala")
Error in fileman_up("tralala") : 
  tralala was not found. Searched up to ./../../../../../../../../../../..
```

See PR #115 for a fix.

cecileherr replied:
Approved and merged

florisvdh reviewed (changes requested):
Nice work! I've tested; works here as well. Independent of the version indeed.

- Note, you always need to update the accompanying Rd files after editing stuff that impacts documentation, and commit those files as well. Some relevant commands and shortcut keys are in `misc/manage_package.R`.
- This and another update are in PR #115. After your review, if approved, you can merge to your branch. After that I'll approve here and merge to dev branch.

florisvdh commented:
> would it be useful to adapt the letter case of the river names (some are all uppercase, others only start with a capital letter)?

Sorry, forgotten to address this (and other opinions welcome of course).

Yes, I think that would be nice indeed. Since this is a function returning the raw data source in some standard format, we don't throw away any records, nor drop actual 'data' columns unless duplicated or irrelevant, nor do we interpret or translate any data. But we still do drop unneeded columns (e.g. length/area if the data is spatial already, or unneeded administrative IDs), we standardize column names, and yes, if data values can be formatted in a better way, that's certainly a nice thing to add.

If you like to add that here, you can always afterwards ask a re-review afterwards with the circular-arrows icon at the top right (![afbeelding](https://user-images.githubusercontent.com/19164640/108391119-a9942180-7211-11eb-93ae-b9d7ae00b4ad.png)).

> Please note I have not added any version control yet on purpose (technically not needed to use v1.7, and I suppose the priority is to make the new layers available to the users - we can tackle the version control at a later stage)

Yes, I follow that too, version handling will need to be reconsidered anyway. We could still have added a pro-forma argument `version` (I had been considering it for #115 but dropped the local commit eventually) and set it to the latest version code by default, but of course we don't use it in this case. Actually such a dummy _was_ added in `read_watersurfaces()` (even as a reason for releasing n2khab 0.3.1), but there it serves to make clear (in the usage section) that the function is not yet adapted to also work with the latest version (which it can't, at least because of the different default filepath: shapefile vs geopackage). It is _needed_ when it actually impacts the function's behaviour.

I have not been too consistent, see below: `watercourse_100mseg` and  `habitatquarries_2020v1` have only one version too. There's a possibility that future version handling will not go through a `version` argument, so shouldn't bother too much, but you can choose what you like.

```bash
$ find . -type f -name *.R -exec grep -nH " version =" {} --colour \;
./R/read_habitatdata.R:122:             version = "habitatmap_stdized_2018_v2"){
./R/read_habitatdata.R:295:             version = "watersurfaces_hab_v3"){
./R/read_habitatdata.R:464:             version = "watersurfaces_v1.0") {
./R/read_habitatdata.R:847:             version = "habitatmap_terr_2018_v2"){
./R/read_habitatdata.R:1178:             version = "habitatsprings_2020v2"){
./R/read_habitatdata.R:1367:             version = "habitatquarries_2020v1"){
./R/read_watercourses.R:105:             version = "watercourse_100mseg_20200807v1"){
```

cecileherr commented:
Adapted the letter case of river_name (2199b78). 
Added the gsub part because I wanted "L'Argentine" while str_to_title gives "L'argentine"

florisvdh reviewed (changes requested):
This is a nice bonus :+1: - it makes the number of factor levels consistent with actual names.

Still an exception to handle: `'De Aa'` vs `De Aa` (v1.6)
A detail: `Ijse` should be `IJse` (v1.7).
If pursuing visual perfection: some strings have double spaces.

There's a more real problem - at least this one needs fixing :wink: :

```r
> read_habitatstreams()
 Error: Problem with `mutate()` input `river_name`.
x could not find function "str_to_title"
ℹ Input `river_name` is `factor(...)`.
Run `rlang::last_error()` to see where the error occurred. 
```

For each function, you should import all functions that you use but are not loaded automatically by base R. Use the roxygen tag `@import` for that; see examples already present in the function. Strictly speaking it's only needed once at package-level (imports are transferred by roxygen to `NAMESPACE`), but this approach makes the functions self-contained. `R CMD check` gives a Note about it, see [here](https://github.com/inbo/n2khab/pull/114/checks?check_run_id=1930876731#step:9:107). (Locally: RStudio -> Build pane -> Check button)

BTW making in-development package available to the current R session for on-the-fly testing: `devtools::load_all()` (or with a button in RStudio, build->more...). Best is to use it each time, after having changed stuff.

cecileherr commented:
Updated the code for read_habitatstreams:
- added minor corrections to river_name (double spaces and correct capital letters for IJ) cc6d179
(maybe a bit overkill but easy to remove if it becomes unnecessary in future versions - I use functions from rstring, do not hesitate to let me know if you prefer base R in the future)
- added the needed @importFrom in read_habitatstreams cc6d179
- added the needed importFrom in NAMESPACE (this one was a gross one :-), sorry ) a09fdbc 

If you don't mind I will neglect this (since I suppose most of the users will rely on the latest version):
> Still an exception to handle: 'De Aa' vs De Aa (v1.6)

florisvdh reviewed (approved):
Well done :+1: thank you!

OK to leave 'De Aa' for now.

Regarding dependencies, I just want to be careful with adding extra _package_ dependencies when it can be easily avoided. `stringr` was already there, so that's fine. Using dependencies makes coding easier, but slows down the loading of the package and it makes future package maintenance a bit harder (external functions change). So I see it as a tradeoff (I would probably be more prudent if I were to start over). An intermediate solution is using a package from 'Suggests' in the Description file, which happens e.g. with `raster` because it adds too much time to load while it's used in just a few functions.
